File: InterviewTraining/js/AlgoHelper/console.js
const TermColors = {
    "HEADER": '\x1B[95m',
    "OKBLUE": '\x1B[94m',
    "OKCYAN": '\x1B[96m',
    "OKGREEN": '\x1B[92m',
    "WARNING": '\x1B[93m',
    "FAIL": '\x1B[91m',
    "ENDC": '\x1B[0m',
    "BOLD": '\x1B[1m',
    "UNDERLINE": '\x1B[4m'
};

const defaultColor = "OKCYAN";

function arrayToStr(array) {
    const retlist = array.map(val => {
        if (Array.isArray(val)) return arrayToStr(val);
        return typeof val === "string" ? val : String(val);
    });
    return "[ " + retlist.join(", ") + " ]";
}

function getColorStr(line, color = defaultColor) {
    if (Array.isArray(line)) line = arrayToStr(line);
    else if (typeof line !== "string") line = JSON.stringify(line);
    return (TermColors[color] || TermColors["FAIL"]) + line + TermColors["ENDC"];
}

function printColor(line, color = defaultColor) {
    console.log(getColorStr(line, color));
}

function scriptHeader(line) {
    const offset = Math.floor(line.length / 2);
    printColor("_".repeat(60), "WARNING");
    printColor(" ".repeat(30 - offset) + line, "WARNING");
    printColor("_".repeat(60), "WARNING");
}

function testHeader(line) {
    const offset = Math.floor(line.length / 2);
    line = `\n${"*".repeat(29 - offset)} ${line} ${"*".repeat(29 - offset)}`;
    printColor(line, "WARNING");
}

function sectionHeader(line) {
    const offset = Math.floor(line.length / 2);
    line = `\n${"-".repeat(29 - offset)} ${line} ${"-".repeat(29 - offset)}`;
    printColor(line, "WARNING");
}

module.exports = {
    TermColors,
    getColorStr,
    printColor,
    scriptHeader,
    testHeader,
    sectionHeader
};

File: InterviewTraining/go/algohelper/console.go
package algohelper

import (
	"fmt"
	"os"
	"strings"
	"golang.org/x/term"
)

// Minimal console helpers to mimic JS coloring utilities.
// Provide ANSI coloring when stdout is a terminal.
var colorMap = map[string]string{
	"HEADER":  "35",
	"OKGREEN": "32",
	"FAIL":    "31",
	"WARNING": "33",
	"BOLD":    "1",
	"OKCYAN":  "36",
}

func isTerminal() bool {
	return term.IsTerminal(int(os.Stdout.Fd()))
}

func colorize(line string, code string) string {
	if !isTerminal() {
		return line
	}
	// allow multiple codes separated by comma
	parts := strings.Split(code, ",")
	return fmt.Sprintf("\x1b[%sm%s\x1b[0m", strings.Join(parts, ";"), line)
}

// PrintColor prints a line (colored when appropriate).
func PrintColor(line string) {
	fmt.Println(line)
}

// PrintColorWith prints a colored line using a color keyword from the map.
func PrintColorWith(line, color string) {
	if code, ok := colorMap[color]; ok {
		fmt.Println(colorize(line, code))
	} else {
		fmt.Println(line)
	}
}

// GetColorStr returns the possibly-colored string; default color when omitted.
func GetColorStr(line string, color ...string) string {
	if len(color) == 0 {
		if code, ok := colorMap["OKCYAN"]; ok {
			return colorize(line, code)
		}
		return line
	}
	if code, ok := colorMap[color[0]]; ok {
		return colorize(line, code)
	}
	return line
}
